src/mitol/google_sheets/api.py

```diff
@@ -0,0 +1,497 @@
+        else sheet_metadata.handler_url_stub()
+    )
+    min_fresh_expiration_date = now + datetime.timedelta(
+        minutes=settings.DRIVE_WEBHOOK_RENEWAL_PERIOD_MINUTES
+    )
+    with transaction.atomic():
+        file_watch, created = GoogleFileWatch.objects.select_for_update().get_or_create(
+            file_id=sheet_file_id,
+            defaults=dict(
+                version=1,
+                channel_id=new_channel_id,
+                activation_date=now,
+                expiration_date=now,
+            ),
+        )
+        if (
+            not created
+            and file_watch.expiration_date > min_fresh_expiration_date
+            and not force
+        ):
+            return file_watch, False, False
+        if file_watch.expiration_date < now:
+            log.error(
+                "Current file watch in the database for %s is expired. "
+                "Some file changes may have failed to trigger a push notification (%s, file id: %s)",
+                sheet_metadata.sheet_name,
+                file_watch,
+                sheet_file_id,
+            )
+        expiration = now + datetime.timedelta(
+            minutes=settings.DRIVE_WEBHOOK_EXPIRATION_MINUTES
+        )
+        try:
+            resp_dict = request_file_watch(
+                sheet_file_id, new_channel_id, handler_url, expiration=expiration
+            )
+        except HttpError as exc:
+            _track_file_watch_renewal(
+                sheet_metadata.sheet_type, sheet_file_id, exception=exc
+            )
+            return None, False, False
+        else:
+            _track_file_watch_renewal(sheet_metadata.sheet_type, sheet_file_id)
+        log.info(
+            "File watch request for push notifications on %s completed. Response: %s",
+            sheet_metadata.sheet_name,
+            resp_dict,
+        )
+        file_watch.activation_date = now
+        file_watch.expiration_date = google_timestamp_to_datetime(
+            resp_dict["expiration"]
+        )
+        file_watch.channel_id = new_channel_id
+        if not created:
+            file_watch.version += 1
+        file_watch.save()
+        return file_watch, created, True
+
+
+def get_sheet_metadata_from_type(sheet_type):
+    """
+    Gets sheet metadata associated with the given sheet type
+
+    Args:
+        sheet_type (str):
+
+    Returns:
+        type(sheets.utils.SheetMetadata): An object with metadata about some sheet type
+
+    Raises:
+         ValueError: Raised if there is no metadata class associated with the given sheet type
+    """
+    if sheet_type == SHEET_TYPE_COUPON_REQUEST:
+        return CouponRequestSheetMetadata()
+    elif sheet_type == SHEET_TYPE_COUPON_ASSIGN:
+        return CouponAssignSheetMetadata()
+    elif sheet_type in {SHEET_TYPE_ENROLL_CHANGE, WORKSHEET_TYPE_REFUND}:
+        return RefundRequestSheetMetadata()
+    raise ValueError(f"No sheet metadata exists matching the type '{sheet_type}'")
```

Comment:
I would alter this to derive from a setting that maps sheet types to the metadata classes. Django's [module_loading](https://docs.djangoproject.com/en/4.0/ref/utils/#module-django.utils.module_loading) util would be a good way to do this.